lib/Bindings/Python/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+################################################################################
+# Set up Python binding tools
+################################################################################
+
+include(AddMLIRPythonExtension)
+add_custom_target(CIRCTBindingsPython)
+
+################################################################################
+# Build native Python extension
+################################################################################
+
+add_mlir_python_extension(CIRCTBindingsPythonExtension _circt
+  INSTALL_DIR
+    python
+  SOURCES
+    CIRCTModule.cpp
+)
+add_dependencies(CIRCTBindingsPython CIRCTBindingsPythonExtension)
+
+################################################################################
+# Python source tree management
+################################################################################
+
+set(PY_SRC_FILES
+  circt/__init__.py
```

Comment:
FYI: Just landed an upstream change to use globs and symlinks for this as it is a better developer flow. Should be mostly copy/paste: https://reviews.llvm.org/rGa9ccdfbc7d73d0abd2506c77efcab1153a82eb35